python/simpler/buffer.py

```
@@ -433,13 +442,26 @@ def materialize(self, desc: BufferDescriptor) -> ImportedBuffer:
         there is no Python path from raw bytes to a descriptor at all, which is what keeps
         ``validate_buffer_descriptor`` a gate rather than a habit.
 
+        Rejects a descriptor that conflicts with the one already mapped under its identity, and a
+        POSIX shm object smaller than the ``nbytes`` its descriptor claims — every view bound check
+        is computed against that number, so an unverified one makes them all vacuous.
+
         Adds no endpoint check of its own: a DEVICE backing resolved here yields a device pointer,
         which is only meaningful on its owner chip. The endpoint × ``address_space`` matrix is a
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)